pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -413,6 +557,15 @@ func (pl *DynamicResources) PreFilter(ctx context.Context, state fwk.CycleState,
 		} else {
 			claimsToAllocate = append(claimsToAllocate, claim)
 
+			if pl.enableExtendedResource {
```

Comment:
I don't think that these `if pl.enableExtendedResource` checks are needed. The following if check is never going to trigger because `s.extended.extendedResourceClaimTplUID` will be empty when the feature is off.

The same applies further down, whenever it is clear that code is not active with the feature disabled and the check is fast.

Removing the feature gate check on graduation then can be done without re-indenting everything.

Reply:
ok, removed the pl.enableExtendedResource check there, and a few other places.